kotest-core/src/commonMain/kotlin/io/kotest/core/runtime/callbacks.kt

```diff
@@ -13,22 +13,41 @@ import io.kotest.core.test.TestResult
 import io.kotest.fp.Try
 import io.kotest.mpp.log
 
+
+fun List<ProjectListener>.resolveName() =
+   groupBy { it.name }
+      .flatMap { entry -> if (entry.value.size > 1) {
+         entry.value.mapIndexed { index, listener -> "${listener.name}_$index" to listener }
+      } else{
+         entry.value.map { it.name to it }
+      }
+      }
+
 /**
  * Invokes any afterProject functions from the given listeners.
  */
-fun List<Listener>.afterProject(): Try<Unit> = Try {
-   log("invokeAfterAll")
-   filterIsInstance<ProjectListener>().forEach { it.afterProject() }
-}.mapFailure { AfterProjectListenerException(it) }
+fun List<Listener>.afterProject(): Try<List<RuntimeException>> = Try {
+   log("invokeAfterProject")
+   filterIsInstance<ProjectListener>()
+      .resolveName()
+      .map { it.first to Try { it.second.afterProject() } }
+      .filter { it.second.isFailure() }
+      .map { AfterProjectListenerException(it.first, (it.second as Try.Failure).error) }
+}.mapFailure { AfterProjectListenerException("afterProjectsInvocation", it) }
 
 /**
  * Invokes the beforeProject listeners, and prints project config using [dumpProjectConfig].
  */
-fun List<Listener>.beforeProject() = Try {
-   log("invokeBeforeAll")
+fun List<Listener>.beforeProject(): Try<List<RuntimeException>> = Try {
```

Comment:
Should this return be Try<list<BeforeProjectListenerException>> ?

Reply:
I don't understand question but try to answer. Earlier beforeProject return just `Try` and don't use success result of it only failed result. It works only until first error. Now we wrap each before\after call with each own `Try` and filter only Failed results because Success results are not interested for us. So return should be only List of exception of known type. 
And Kotlin compiler doesn't allow to use just list without type `Try<List>` 
![image](https://user-images.githubusercontent.com/12140191/79889551-dcf5fc00-8406-11ea-9333-05f560276c7e.png)

Reply:
Github hid some of the message because of the < > tags. I was asking if the return type should be List of AfterProjectListenerException rather than List of RuntimeException